projects/plugins/boost/app/assets/src/js/utils/generate-critical-css.ts

```diff
@@ -205,11 +228,31 @@ async function generateForKeys(
 					passthrough,
 				} );
 
+				// Track showstopper Critical CSS generation error.
+				const eventProps = {
+					time: Date.now() - startTime,
+					provider_key: providerKey,
+					error_message: err.message,
+					error_type: err.type,
+				};
+				jpTracksAJAX.record_ajax_event( 'critical_css_failure', 'click', eventProps );
+
 				return;
 			}
 		}
 	}
 
+	// Track complete Critical CSS generation result.
+	const eventProps = {
+		time: Date.now() - startTime,
+		block_count: stepsPassed,
+		error_count: stepsFailed,
+		average_size: totalSize / stepsPassed,
```

Comment:
```suggestion
		average_size: totalSize / Math.max( 1, stepsPassed ),
```

... just to be cautious and avoid divide-by-zero errors.